public_html/wp-content/plugins/wcpt/wcpt-event/class-event-admin.php

```diff
@@ -330,16 +330,29 @@ public static function schedule_decline_notification( $event, $label, $location
 	 * @param int    $event_id
 	 * @param string $label
 	 * @param string $location
+	 *
+	 * @return null|bool|string
 	 */
 	public static function send_decline_notification( $event_id, $label, $location ) {
+		$declined_notification_key = 'Sent declined notification';
+		$already_notified          = get_post_meta( $event_id, $declined_notification_key, true );
+		if ( $already_notified ) {
+			return null;
+		}
+
 		$message = sprintf(
 			'A %s application for %s has been declined, and the applicant has been informed via email.',
 			$label,
 			$location
 		);
 
 		$attachment = create_event_status_attachment( $message, $event_id, '' );
-		wcpt_slack_notify( COMMUNITY_TEAM_SLACK, $attachment );
+
+		$notification_sent = wcpt_slack_notify( COMMUNITY_TEAM_SLACK, $attachment );
+		if ( $notification_sent ) {
+			update_post_meta( $event_id, $declined_notification_key, true );
+		}
+		return $notification_sent;
```

Comment:
I'm wondering if this should return anything, since it's a cron callback. The return value won't ever be used, and it feels a bit odd for a function to return `null` or a `bool` or a `string`.

What's the goal/benefit of tracking that? If we want to know about failures at some point, it seems like adding logging to `wcpt_slack_notify()` might be a better place for them. If it's about unit-testability, we'd probably need to mock `wcpt_slack_notify()` anyway, so we could do that in a way that lets us test for success/failure (similar to how PHPMailer is mocked by Core's test suite).

Reply:
no good reason, so it returns null|bool|string because it returns whatever the response is from the web request, which could be anything really ( and null in case of connection error ) . i like keeping return values documented if a function can return something in case we may need that in future, but right we are not using it.
lets keep it for now, because there is no overhead in keeping it, but you are correct that its very unlikely that we will use it.